gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/tasks/AbstractJPackageTask.kt

```diff
@@ -276,9 +284,10 @@ abstract class AbstractJPackageTask @Inject constructor(
 
         if (targetFormat != TargetFormat.AppImage) {
             // Args, that can only be used, when creating an installer
-            cliArg("--app-image", appImage)
+            cliArg("--app-image", appImage.dir("${packageName.get()}.app"))
```

Comment:
This change looks suspicious. With non app store packaging previous value clearly worked.
I'm not sure, why this is required and whether it could break cases, that worked previously.
I don't doubt, that you've had a reason to add it, but I would like to double check it before merging.

Reply:
Without this change I get the following error:
```
...
[16:18:13.997] Running /usr/bin/productbuild
[16:18:14.015] Command [PID: 6897]:
    /usr/bin/productbuild --resources /var/folders/61/dv805tx974q7kldwx81tf9b80000gn/T/jdk.jpackage12469847631489678324/config --sign 3rd Party Mac Developer Installer: Thomas Vos (4J6Y3WBQB6) --product /var/folders/61/dv805tx974q7kldwx81tf9b80000gn/T/jdk.jpackage12469847631489678324/config/product-def.plist --component /var/folders/61/dv805tx974q7kldwx81tf9b80000gn/T/jdk.jpackage12469847631489678324/root-6402241906218587642/Hue Essentials.app /Applications /Users/thomas/AndroidStudioProjects/HueEssentials/desktop/build/compose/binaries/main/pkg/Hue Essentials-1.0.0.pkg
[16:18:14.015] Output:
    productbuild: error: The component at "/var/folders/61/dv805tx974q7kldwx81tf9b80000gn/T/jdk.jpackage12469847631489678324/root-6402241906218587642/Hue Essentials.app" does not exist or is not a directory.
[16:18:14.015] Returned: 1
...
```

By adding packageName.app to the --app-image option it works fine. Not sure why this is needed.

I have not tested the code on other platforms, but this will fail because ".app" is macOS specific. I can add this within a condition to only add this for macOS and macAppStore.

Reply:
What do you think of https://github.com/JetBrains/compose-jb/pull/1613/commits/b24cd0ddaba5ff4810f94b25caa9d7b92602aa84?

Are you able to test this? You could still exclude this change of course, but I think that would make the PR pointless as for me this is needed for it to work. Let me know what you think.

Reply:
@AlexeyTsvetkov FYI, when creating a pkg using packagePkg, without this PR, after installation on my mac it is installed like this:
![Screenshot 2022-01-14 at 20 33 57](https://user-images.githubusercontent.com/11077217/149574580-866e3341-e243-4c18-8531-2b3fc42718c4.png)

Think there is indeed an issue in compose. But maybe it's only for the newer JDK version, not sure. This was with JDK 17.

Reply:
Created issue here: https://github.com/JetBrains/compose-jb/issues/1712